src/main/settings/custom_settings.cpp

```diff
@@ -1149,7 +1149,7 @@ void TempDirectorySetting::SetGlobal(DatabaseInstance *db, DBConfig &config, con
 	if (!config.options.enable_external_access) {
 		throw PermissionException("Modifying the temp_directory has been disabled by configuration");
 	}
-	config.options.temporary_directory = input.ToString();
+	config.options.temporary_directory = input.IsNull() ? "" : input.ToString();
```

Comment:
Should ToString() throw on IsNull?
And then have a `ToString(null_string)` return null_string on null?

This seems a pretty reasonable mistake.